### ISSUE TITLE
Client broken state with stalwart mailserver

### ISSUE THREAD

**Issue by SinnySupernova** [linagora/tmail-flutter#3359]
### Description
I'm trying to use tmail web with the [stalwart mailserver](https://github.com/stalwartlabs/mail-server). Unfortunately, there seem to be a few issues right off the bat:
- layout is misplaced (see screenshot below)
- client makes request to `https://mail.tld/.well-known/webfinger?resource=https://mail.tld&rel=http://openid.net/specs/connect/1.0/issuer`, which fails (cors failure when ran from the browser client, and 404 when ran with curl)
If this is required by the (jmap?) spec I'll open an issue in stalwart, if not - there needs to be a workaround, e.g. setting env vars. Either way the client should not end up in such a state.

### Expected result
Web client doesn't get into completely broken state :sweat_smile: 

### Current behavior
![Screenshot_20241217_144721](https://github.com/user-attachments/assets/13675b18-4dc5-442c-bb97-f273defa871d)

yes,this is all I see and neither the button nor the privacy policy links are clickable

### Preconditions (optional)
n/a

#### Reproduction Steps
- Open tmail web in librewolf (=firefox)

### Acceptance criteria
??

### Context

firefox on linux

### Additional information
n/a

**Comment by chibenwa** [comment]
Hello @SinnySupernova , thanks for report!

> If this is required by the (jmap?) spec I'll open an issue in stalwart

No not realy JMAP spec do not state anything about auth...

> client makes request to https://mail.tld/.well-known/webfinger?resource=https://mail.tld&rel=http://openid.net/specs/connect/1.0/issuer, which fails (cors failure when ran from the browser client, and 404 when ran with curl)

The right thing here is to correctly handle these errors in the front and fallback to regular (basic auth) login page.

Otherwise we can document a NGINX setup in front of the JMAP endpoint actually compatible with basic auth.

Finally @florentos17 you might run into this issue when running on top of Cyrus so it might be worth a look.

**Comment by chibenwa** [comment]
> The right thing here is to correctly handle these errors in the front and fallback to regular (basic auth) login page.

We need to be careful in differenciating 'outage' ie 500, timeouts where saying oups is ok
From unconfigured endpoint where we should display login page.

Contribution appreciated if in your reach.

**Comment by SinnySupernova** [comment]
Can we have the client completely bypass that webfinger call? Or even better - specify an environmental variable that points to the correct issuer URL. 
I would like to avoid having to add an extra nginx container to the stalwart deployment (it is using a different reverse proxy)

Additionally, I'd like to share a few extra things I just checked:

- v0.13.1 doesn't call the webfinger endpoint and shows the email/password fields
- both latest and v0.13.1 layout is completely unusable in my librewolf browser **because it has WebGL disabled**

v0.13.1 looks like this:
![Screenshot_20241217_174253](https://github.com/user-attachments/assets/9322d2bc-5c87-4da4-b4c7-6fd8e50b776b)

- apparently you _can_ click the elements where they _should have been_, but visually everything is shifted upwards (see where the cursor is, and the slightly darker color of the password field)

I'm looking into [the docs](https://docs.flutter.dev/platform-integration/web/renderers) for options. Will try to build with --wasm and see where it takes me.

**Comment by chibenwa** [comment]
> specify an environmental variable that points to the correct issuer URL.

For web why not but for mobile app loading per deployment config is not relevant.

> both latest and v0.13.1 layout is completely unusable in my librewolf browser because it has WebGL disabled

Old version, and sadly we are also tied down to flutter limitations

**Comment by TheDarkula** [comment]
First off, thank you for all of your work!

I am also running a stalwart instance, and it loads correctly with the android client.

I just tried the `tmail-web` docker image, and I have noticed a few issues.

One, setting the `SERVER_URL` currently needs to be written to `/usr/share/nginx/html/assets/env.file`, which defeats the point of using environment variables with docker images.
It would make more sense for the server to just pick up the `SERVER_URL` value from the environment.

Secondly, the web server tries to connect to `.well-known/webfinger`, but stalwart does not have this endpoint.
It listens on `.well-known/jmap`, which the android client seems to be happy with.


Also, I have this traefik middleware:

```
spec:
  headers:
    accessControlAllowMethods:
      - "GET"
      - "OPTIONS"
      - "PUT"
      - "POST"
      - "DELETE"
    accessControlAllowOriginList:
      - "*"
    accessControlAllowHeaders:
      - "*"
```

but when accessing tmail-web, the web console shows a CORS error still:

```
Access to XMLHttpRequest at 'https://jmap.domain.com/.well-known/webfinger?resource=https://jmap.domain.com&rel=http://openid.net/specs/connect/1.0/issuer' from origin 'https://webmail.domain.com' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.
```

**Comment by chibenwa** [comment]
> It would make more sense for the server to just pick up the SERVER_URL value from the environment.

Feel free to contribute this. We could interpolate the value in the NGINX conf upon boot.

> It would make more sense for the server to just pick up the SERVER_URL value from the environment.

This is used by TMail for OIDC discovery.

Hopefully a 404 on this endpoint shall be treated as "We do not support OIDC here" and the client must then continue its life with basic auth.

@hoangdat can you verify the client behaviour in such case? And fix it please?

> but when accessing tmail-web, the web console shows a CORS error still

Sorry I'm not a traefik dev.

**Comment by TheDarkula** [comment]
I can have a look and see about making that change.

I have also tried this with [Static Web Server](https://static-web-server.net/) and it gives the same error.

**Comment by chibenwa** [comment]
Yes static webserver does the same than mobile.

**Comment by TheDarkula** [comment]
@chibenwa No, I mean a tried running tmail through SWS, and it also gives the same errors as traefik.

**Comment by jkaberg** [comment]
Any updates in terms of the CORS or webfinger issues?

**Comment by theeddie** [comment]
update
